modules/tensor_mechanics/src/materials/RankTwoCylindricalComponent.C

```diff
@@ -0,0 +1,60 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "RankTwoCylindricalComponent.h"
+#include "RankTwoScalarTools.h"
+
+registerMooseObject("TensorMechanicsApp", RankTwoCylindricalComponent);
+
+InputParameters
+RankTwoCylindricalComponent::validParams()
+{
+  InputParameters params = Material::validParams();
+  params.addClassDescription("Compute a cylindrical stress/strain property of a RankTwoTensor");
+  params.addRequiredParam<MaterialPropertyName>("rank_two_tensor",
+                                                "The rank two material tensor name");
+  params.addParam<std::string>("calculation_name",
+                               "Optional parameter that allows the user to define "
+                               "multiple mechanics material systems on the same "
+                               "block, i.e. for multiple phases");
+  params.addParam<MooseEnum>(
+      "scalar_type", RankTwoScalarTools::cylindricalOptions(), "Type of scalar output");
```

Comment:
`scalar_type` should be renamed to something more descriptive like `cylindrical_component` or just `component`